src/datajudge/formatter.py

```diff
@@ -0,0 +1,50 @@
+import abc
+import re
+
+from colorama import Back
+
+STYLING_CODES = r"\[(numMatch|numDiff)\](.*?)\[/\1\]"
+
+
+class Formatter(abc.ABC):
+    @abc.abstractmethod
+    def fmt_str(self, result: str) -> str:
+        pass
+
+
+class DefaultFormatter(Formatter):
+    def __init__(self):
+        self.known_bb_pattern = re.compile(STYLING_CODES)
+
+    # Just ignore styling in the default formatter
+    def apply_formatting(self, _: str, inner: str) -> str:
+        return inner
+
+    def fmt_str(self, string: str) -> str:
+        # Replace codes with platform specific styling
+        string = self.known_bb_pattern.sub(
+            lambda m: self.apply_formatting(m.group(1), m.group(2)), string
+        )
+
+        return string
+
+
+class AnsiColorFormatter(DefaultFormatter):
+    def __init__(self):
```

Comment:
I don't think you need this here. At least you're not using it in `HtmlFormatter`.